src/ui/public/management/section.js

```diff
@@ -0,0 +1,68 @@
+import {assign} from 'lodash';
+import IndexedArray from 'ui/indexed_array';
+
+export default class ManagementSection {
+
+  /**
+   * @param {string} id
+   * @param {object} options
+   * @returns {ManagementSection}
+   */
+
+  constructor(id, options = {}) {
+    this.display = id;
+    this.id = id;
+    this.items = new IndexedArray({
+      index: ['id'],
+      order: ['order']
+    });
+
+    if (!options.url) {
+      options.url = '#/management/' + options.path || '';
+    }
+
+    assign(this, options);
+  }
+
+  /**
+   * Registers a section
+   *
+   * @param {string} id
+   * @param {object} options
+   * @returns {ManagementSection}
+   */
+
+  register(id, options = {}) {
+    const item = new ManagementSection(id, assign(options, { parent: this }));
+
+    if (this.hasItem(id)) {
+      throw new Error(`'${id}' is already registered`);
```

Comment:
Another reason to avoid exposing the class (or using classes at all perhaps): I could accidentally circumvent this check by writing

```
new ManagementSection('existingId', { parent: rootManagementSection });
```